runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -1224,17 +1652,11 @@ TR::Register *J9::Power::TreeEvaluator::awrtbariEvaluator(TR::Node *node, TR::Co
    return NULL;
    }
 
-TR::Register *J9::Power::TreeEvaluator::irdbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-   {
-   return TR::TreeEvaluator::irdbariEvaluator(node, cg);
-   }
-
-TR::Register *J9::Power::TreeEvaluator::irdbariEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+TR::Register *iGenerateSoftwareReadBarrier(TR::Node *node, TR::CodeGenerator *cg)
    {
-   TR_ASSERT(TR::Compiler->om.readBarrierType() != gc_modron_readbar_none, "iReadBarrierEvaluator");
-#ifdef OMR_GC_CONCURRENT_SCAVENGER
-   TR_ASSERT(cg->comp()->useCompressedPointers(), "irdbarEvaluator is expecting compressed references");
-
+#ifndef OMR_GC_CONCURRENT_SCAVENGER
+   TR_ASSERT_FATAL(false, "Concurrent Scavenger not supported.");
```

Comment:
I think you need a "return NULL" after the assert fatal statement.

Reply:
Yes, I agree, I think we discussed that, but somehow it compiled in 32-bit mode...

Reply:
TR_ASSERT_FATAL calls exit() which does not return. So a return would be unreachable code.

Reply:
I don't want to prolong this discussion much more but, for the future, I would argue for  adding returns. By the nature of macros, they can be changed or undefined. I think we should nor rely on what is inside them.

Reply:
Let's ask @0xdaryl :)

Reply:
Are you presently getting a compiler warning for a missing return?  I doubt the "fatal" semantics of the fatal assert macro are going to change very much which would introduce a problem down the road.  The developer's choice of a fatal assert means something very wrong has happened or will happen imminently if we don't abort.  For now, I'm comfortable with the cleaner looking code without the return.

If we were to require one here, there are many other fatal asserts in the OpenJ9 and OMR code base that we would need to give consideration to as well.